nexus/src/nexus.rs

```diff
@@ -3660,25 +3660,25 @@ impl Nexus {
         &self,
         opctx: &OpContext,
         silo_user_id: Uuid,
-    ) -> LookupResult<SiloUser> {
-        let (.., db_silo_user) = LookupPath::new(opctx, &self.db_datastore)
-            .silo_user_id(silo_user_id)
-            .fetch()
-            .await?;
-        Ok(db_silo_user)
+    ) -> LookupResult<(authz::SiloUser, db::model::SiloUser)> {
+        let (.., authz_user, db_silo_user) =
+            LookupPath::new(opctx, &self.db_datastore)
```

Comment:
Is this change because the silo integration test needs it?  If so, does it still after the other recent changes?

This isn't wrong but it's different than all the other fetch functions in Nexus so it'd be nice to better understand why and whether this belongs somewhere else.  For example, if this is a test-only interface, it probably belongs in NexusTestInterfaces instead.

Reply:
No, it's used in all the HTTP endpoint functions for keys (e.g., `sshkeys_get`) because we now pass an `authz::SiloUser` to, e.g., `Nexus::ssh_keys_list` &c. It could be a separate function, e.g., `silo_user_authz` or something.

Reply:
Ah, I see!  So the pattern used for most other resources is:

- the HTTP endpoint functions do very little -- they pass their Uuid/Name/String arguments to analogous Nexus functions (without having done any kind of database lookup)
- the Nexus functions do the translation from Uuid/Name/etc. to `authz::Foo` using the LookupPath stuff
- the Datastore functions accept the `authz::Foo` versions

(as a corollary: `foo_fetch()` is only used by the "GET" HTTP endpoint, so it _just_ returns the db object)

There's some downsides to this but it seems worth doing the same thing here unless there's a reason not to.  What do you think?

Reply:
Ok, I think I see. So the Nexus functions should take the Silo user ID and whatever ssh-specific params they need, look up the `authz::SiloUser` internally, then pass that down to the datastore level, right? Should be no problem.

Reply:
Fixed in 79d482a1. _Much_ nicer, thank you for the helpful explanation.